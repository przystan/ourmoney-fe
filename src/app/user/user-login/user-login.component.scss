:host {
  display: flex;
  flex-direction: column;
}

.login-title {
  opacity: 0.7;
  font-weight: 400;
  font-size: 20pt;
  font-family: 'Barlow Semi Condensed', sans-serif;
  text-transform: uppercase;
  margin-bottom: 0.7em;
}

.login-btn {
  margin-top: 0.83rem;
  text-transform: uppercase;
  font-size: 0.7rem;
  letter-spacing: 0.1em;
}

.lock-icon {
  font-size: 16pt !important;
  margin-bottom: 0.1rem;
}

.description {
  opacity: 0.7;
  font-size: 0.7rem;
  margin-top: 0.5em;

  a {
    font-weight: bold;
    color: var(--button-color);
    cursor: pointer;
  }
}
