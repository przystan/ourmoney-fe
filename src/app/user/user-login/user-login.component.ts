import { Component, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-user-login',
  template: `
    <h1 class="login-title">Login</h1>
    <mat-form-field appearance="legacy" floatLabel="never">
      <mat-label>Username</mat-label>
      <input matInput placeholder="Your username" />
      <mat-icon matPrefix>person</mat-icon>
    </mat-form-field>
    <mat-form-field appearance="legacy" floatLabel="never">
      <mat-label>Password</mat-label>
      <mat-icon matPrefix>lock</mat-icon>
      <input
        matInput
        placeholder="Password"
        [type]="isPasswordHidden ? 'password' : 'text'"
      />
      <button
        mat-icon-button
        matSuffix
        (click)="isPasswordHidden = !isPasswordHidden"
        [attr.aria-label]="'Hide password'"
        [attr.aria-pressed]="isPasswordHidden"
      >
        <mat-icon class="lock-icon" inline="true">{{
          isPasswordHidden ? 'visibility_off' : 'visibility'
        }}</mat-icon>
      </button>
    </mat-form-field>
    <button mat-raised-button color="accent" type="submit" class="login-btn">
      Login
    </button>
    <div class="description">Don't have an account? <a>Sign up</a></div>
  `,
  styleUrls: ['./user-login.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class UserLoginComponent {
  isPasswordHidden = true;
}
