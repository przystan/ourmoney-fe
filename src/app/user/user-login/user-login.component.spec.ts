import { Spectator, createComponentFactory } from '@ngneat/spectator';
import { UserLoginComponent } from './user-login.component';

describe('UserLoginComponent', () => {
  let spectator: Spectator<UserLoginComponent>;

  const createComponent = createComponentFactory({
    component: UserLoginComponent
  });

  beforeEach(() => {
    spectator = createComponent();
  });

  it('should match snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });
});
