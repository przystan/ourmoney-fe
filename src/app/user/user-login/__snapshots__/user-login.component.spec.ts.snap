// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLoginComponent should match snapshot 1`] = `
<app-user-login
  isPasswordHidden={[Function Boolean]}
>
  <h1
    class="login-title"
  >
    Login
  </h1><mat-form-field
    appearance="legacy"
    floatlabel="never"
  >
    <mat-label>
      Username
    </mat-label>
    <input
      matinput=""
      placeholder="Your username"
    />
    <mat-icon
      matprefix=""
    >
      person
    </mat-icon>
  </mat-form-field><mat-form-field
    appearance="legacy"
    floatlabel="never"
  >
    <mat-label>
      Password
    </mat-label>
    <mat-icon
      matprefix=""
    >
      lock
    </mat-icon>
    <input
      matinput=""
      placeholder="Password"
      type="password"
    />
    <button
      aria-label="Hide password"
      aria-pressed="true"
      mat-icon-button=""
      matsuffix=""
    >
      <mat-icon
        class="lock-icon"
        inline="true"
      >
        visibility_off
      </mat-icon>
    </button>
  </mat-form-field><button
    class="login-btn"
    color="accent"
    mat-raised-button=""
    type="submit"
  >
     Login 
  </button><div
    class="description"
  >
    Don't have an account? 
    <a>
      Sign up
    </a>
  </div>
</app-user-login>
`;
